src/internal/db/engine.py

```
@@ -66,3 +79,118 @@ def session_scope():
+    global _async_engine
+    if _async_engine is None:
+        with _async_engine_lock:
+            if _async_engine is None:
+                _async_engine = create_async_engine_from_env()
+    return _async_engine
+
+
+def get_async_session_factory() -> async_sessionmaker[AsyncSession]:
+    """Get or create the singleton async sessionmaker."""
+    global _async_session_factory
+    if _async_session_factory is None:
```

Pylint (4.0.5)

[error] Using variable '_async_session_factory' before assignment

(E0601)